docs/api/session.md

```diff
@@ -831,6 +831,21 @@ this session just before normal `preload` scripts run.
 Returns `String[]` an array of paths to preload scripts that have been
 registered.
 
+#### `ses.setCodeCachePath(path)`
+
+* `path` String - The Code cache location.
```

Comment:
What does setting the code cache path mid-session do? Does it clear the cache? What happens if `path` is empty? Does `path` need to be an existing folder? What happens if it isn't? What happens if the code cache path can't be created (e.g. due to lack of permissions, or because a parent path element is a file instead of a directory)?

Reply:
> What does setting the code cache path mid-session do? Does it clear the cache?

This is not error prone, the old cache path will remain on disk and will not be used. This is in way of clearing the code caches, since next time the files are compiled they don't have corresponding caches in the new path and will be written post compilation.

> What happens if path is empty? Does path need to be an existing folder? What happens if it isn't? 

Path cannot be empty but it need not be an existing folder, the runtime will create one if it does not exist. I can tweak the api to throw an error for empty path values.

> What happens if the code cache path can't be created (e.g. due to lack of permissions, or because a parent path element is a file instead of a directory)?

It uses the simple disk cache backend and if it fails to initialize then the write and fetch calls to the backend will silently fail. That is no code cache will be used. for ex: https://source.chromium.org/chromium/chromium/src/+/main:content/browser/code_cache/generated_code_cache.cc;l=358-361

There is also a comment that the cache backend creator will delete the contents of the folder when it is unable to initialize in that path. I need to confirm this.

Reply:
ok, seems reasonable to silently fail if the path is invalid. probably document that though :)

Reply:
@ckerr the behavior aligns with how we allow the user to change the user data directory via `app.getPath`, the application should be responsible for clearing any stale cache/data directories it had created.